cterasdk/core/files/collaboration.py

```diff
@@ -0,0 +1,255 @@
+        if account.is_local:
+            if account.account_type == PortalAccountType.User and collaborator.type == CollaboratorType.LU:
+                logging.getLogger().debug('Found local user account. %s', {'name': account.name})
+                return collaborator
+            if account.account_type == PortalAccountType.Group and collaborator.type == CollaboratorType.LG:
+                logging.getLogger().debug('Found local group account. %s', {'name': account.name})
+                return collaborator
+            logging.getLogger().warning('Expected to find local %s but found domain %s', account.account_type, account.account_type)
+        else:
+            if account.account_type == PortalAccountType.User and collaborator.type == CollaboratorType.DU:
+                logging.getLogger().debug('Found domain user account. %s', {'name': account.name, 'domain': account.directory})
+                return collaborator
+            if account.account_type == PortalAccountType.Group and collaborator.type == CollaboratorType.DG:
+                logging.getLogger().debug('Found domain group account. %s', {'name': account.name, 'domain': account.directory})
+                return collaborator
+            logging.getLogger().warning('Expected to find domain %s but found local %s', account.account_type, account.account_type)
+    return None
+
+
+def remove_share_recipients(ctera_host, path, accounts):
+    share_info = get_share_info(ctera_host, path)
+    current_accounts = _obtain_current_accounts(share_info)
+
+    accounts_to_keep, accounts_to_remove = [], []
+    if current_accounts:
+        for idx, current_account in enumerate(current_accounts):
+            if current_account not in accounts:
+                accounts_to_keep.append(share_info.shares[idx])
+            else:
+                logging.getLogger().debug('Found recipient to remove. %s', {'account': str(current_account)})
+                accounts_to_remove.append(current_account)
+        if not accounts_to_remove:
+            logging.getLogger().info(
+                'Could not find share recipients to remove. %s',
+                {
+                    'path': str(path.relativepath),
+                    'recipients': [str(member) for member in accounts]
+                }
+            )
+        else:
+            share_param = _create_share_param(path.fullpath(), share_info.teamProject, share_info.allowReshare, share_info.shouldSync)
+            _update_share_param(share_param, accounts_to_keep)
+            logging.getLogger().info(
+                'Removing share recipients. %s',
+                {'path': str(path.relativepath), 'recipients': [str(account) for account in accounts_to_remove]}
+            )
+            ctera_host.execute('', 'shareResource', share_param)
+    else:
+        logging.getLogger().info('Share has no recipients. %s', {'path': str(path.relativepath)})
+    return accounts_to_remove
+
+
+def get_share_info(ctera_host, path):
```

Comment:
1. What happens when the path does not exist? Do you get an error response that will trigger an exception?
2. What happens when the path exists, but there are no shares? If it returns an empty share list, will the "add_share_recipients" method function as an incomplete "share"

Reply:
1) Code will throw a "RemoteDirectoryNotFound" exception
2) Server will return a 500 "Resource does not exist"

Reply:
OK, great (through it should not return 500, it is the Portal's issue and not the SDK's)